neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt

```diff
@@ -806,26 +806,43 @@ object Neo4jStoreSingleton : GraphStore {
         }
 
         return IO {
-            Either.monad<StoreError>()
-                    .binding {
-                        simProfiles.bind().map { simProfile ->
-                            val regionId = (regionCode ?: map[simProfile.id])
-                                    ?: ValidationError(type = simProfileEntity.name, id = simProfile.iccId, message = "SimProfile not linked to any region")
-                                            .left()
-                                            .bind()
-
-                            val simEntry = simManager.getSimProfile(
-                                    hlr = hssNameLookup.getHssName(regionId),
-                                    iccId = simProfile.iccId)
-                                    .mapLeft { NotFoundError(type = simProfileEntity.name, id = simProfile.iccId) }
-                                    .bind()
-                            org.ostelco.prime.model.SimProfile(
-                                    iccId = simProfile.iccId,
-                                    alias = simProfile.alias,
-                                    eSimActivationCode = simEntry.eSimActivationCode,
-                                    status = simEntry.status)
-                        }
-                    }.fix()
+            Either.monad<StoreError>().binding {
+                simProfiles.bind().map { simProfile ->
+
+                    val regionId = (regionCode ?: map[simProfile.id])
+
+                    val simEntry = if (regionId != null) {
+                        simManager.getSimProfile(
+                                hlr = hssNameLookup.getHssName(regionId),
+                                iccId = simProfile.iccId).fold(
+                                { error ->
+                                    logger.warn("SimProfile not found in SIM Manager DB. region: {}, iccId: {}, error: {}", regionId, simProfile.iccId, error)
+                                    SimEntry(
+                                            iccId = simProfile.iccId,
+                                            status = NOT_READY,
+                                            eSimActivationCode = "Dummy eSIM",
+                                            msisdnList = emptyList()
+                                    )
+                                },
+                                { it }
+                        )
+                    } else {
+                        logger.warn("SimProfile not linked to any region. iccId: {}", simProfile.iccId)
+                        SimEntry(
+                                iccId = simProfile.iccId,
+                                status = NOT_READY,
+                                eSimActivationCode = "Dummy eSIM",
+                                msisdnList = emptyList()
+                        )
+                    }
+
+                    org.ostelco.prime.model.SimProfile(
```

Comment:
Why can't the original simProfile instance be used? This is just a copy? If it's just a copy, why not use a .copy() method?

Reply:
Because they are instances of different classes.
https://github.com/ostelco/ostelco-core/blob/develop/model/src/main/kotlin/org/ostelco/prime/model/Entities.kt#L352-L356

https://github.com/ostelco/ostelco-core/blob/develop/neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/model/Model.kt#L29-L35

Reply:
The classes are different because we have cases where a single external model class is split into 2 classes in the Graph DB model - one becomes Entity Node and other is Relation.

Reply:
In this case, the external class has an extra field compared to the internal Graph DB model class since values for the extra field(s) comes from SIM Manager DB.